step-framework-timeseries/src/main/java/step/core/timeseries/TimeSeries.java

```diff
@@ -1,60 +1,128 @@
 package step.core.timeseries;
 
-import step.core.collections.Collection;
-import step.core.collections.CollectionFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import step.core.collections.Filter;
+import step.core.collections.Filters;
 import step.core.collections.IndexField;
+import step.core.collections.SearchOrder;
 import step.core.timeseries.aggregation.TimeSeriesAggregationPipeline;
 import step.core.timeseries.bucket.Bucket;
+import step.core.timeseries.ingestion.TimeSeriesIngestionPipeline;
+import step.core.timeseries.ingestion.TimeSeriesIngestionPipelineSettings;
 import step.core.timeseries.query.TimeSeriesQuery;
 
+import java.io.Closeable;
+import java.io.IOException;
 import java.util.*;
-import java.util.stream.Collectors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
 
-public class TimeSeries {
+import static step.core.timeseries.TimeSeriesConstants.TIMESTAMP_ATTRIBUTE;
 
-    private final Collection<Bucket> collection;
-    private final Integer timeSeriesResolution;
+public class TimeSeries implements Closeable {
 
-    public TimeSeries(Collection<Bucket> collection,  Integer timeSeriesResolution) {
-        this.collection = collection;
-        this.timeSeriesResolution = timeSeriesResolution;
+    private static final Logger logger = LoggerFactory.getLogger(TimeSeries.class);
+
+    private final List<TimeSeriesCollection> handledCollections;
+    private final TimeSeriesAggregationPipeline aggregationPipeline;
+    
+
+    TimeSeries(List<TimeSeriesCollection> handledCollections) {
+        this.handledCollections = handledCollections;
+        aggregationPipeline = new TimeSeriesAggregationPipeline(handledCollections);
     }
 
-    public TimeSeries(CollectionFactory collectionFactory, String collectionName, Integer ingestionResolutionPeriod) {
-        this(collectionFactory.getCollection(collectionName, Bucket.class), ingestionResolutionPeriod);
+    public void ingestDataForAllResolutions() {
```

Comment:
This method seems to be  used to ingest data for new resolutions, make it clear with the name and javadoc